baseplate/experiments/__init__.py

```diff
@@ -134,25 +138,43 @@ def variant(self, name, user=None, bucketing_event_override=None,
         :rtype: :py:class:`str`
         :return: Variant name if a variant is active, None otherwise.
         """
+
         experiment = self._get_experiment(name)
 
         if experiment is None:
             return None
 
         inputs = dict(kwargs)
+
+        """ clean up user id. If not a baseplate user, we risk inconsistency."""
         if user:
             if user.is_logged_in:
                 inputs["user_id"] = user.id
                 inputs["user_roles"] = user.roles
+                inputs["logged_in"] = user.is_logged_in
             else:
                 inputs["user_id"] = user.loid
-            inputs["logged_in"] = user.is_logged_in
```

Comment:
Why this change?

Reply:
can we be certain that user.is_logged_in is always populated and never null?

Reply:
Yes https://github.com/reddit/baseplate/blob/master/baseplate/core.py#L262

Reply:
gotcha. I think this actual change was a carryover from my various modifications that I didn't "correct" since the new change isn't broken. Will fix.